internal/app/server/open_saves.go

```diff
@@ -250,6 +250,45 @@ func (s *openSavesServer) QueryRecords(ctx context.Context, req *pb.QueryRecords
 	}, nil
 }
 
+func (s *openSavesServer) GetRecords(ctx context.Context, req *pb.GetRecordsRequest) (*pb.GetRecordsResponse, error) {
+	records, err := s.metaDB.GetRecords(ctx, req.GetStoreKeys(), req.GetKeys())
+
+	// Check if there was an unexpected error
+	var errors metadb.MultiError
+	if err != nil {
+		var ok bool
+		if errors, ok = err.(metadb.MultiError); !ok {
+			log.Errorf("GetRecords unable to retrieve records: %v", err)
+			return nil, err
+		}
+	}
+
+	response := &pb.GetRecordsResponse{
+		Results: []*pb.GetRecordsResponse_Result{},
+	}
+	for i, rec := range records {
+		var protoRec *pb.Record
+		var storeKey string
+		var statusCode codes.Code
+
+		if rec != nil {
+			protoRec = rec.ToProto()
+			storeKey = rec.StoreKey
+		}
+		if errors != nil {
+			statusCode = status.Code(errors[i])
```

Comment:
Do we have to redefine `datastore.MultiError`?

What do you think about having `metadb.GetRecords` return datastore's `MultiError` and then call `datastoreErrToGRPCStatus` in this loop?

Reply:
I was trying to address a suggestion from Emma to not expose the datastore.MultiError in open_saves.go, but I think this is really overkill and add unnecessary complexity for what it's worth.
I will be working on removing the redefinition of this type in metadb and rely on the original datastore.MultiError.

Reply:
I still prefer to convert the errors from datastore in the metadb module to keep things consistent.